lib/rex/proto/rfb/constants.rb

```diff
@@ -13,39 +13,69 @@
 ##
 
 module Rex
-module Proto
-module RFB::Constants
+  module Proto
+    module RFB::Constants
+      DefaultPort = 5900
 
-DefaultPort = 5900
+      # Version information
+      MajorVersions = [3, 4, 5]
+      # NOTE: We will emulate whatever minor version the server reports.
 
-# Version information
-MajorVersions = [3, 4]
-# NOTE: We will emulate whatever minor version the server reports.
+      # Security types https://datatracker.ietf.org/doc/html/rfc6143#section-8.1.2
+      # https://www.iana.org/assignments/rfb/rfb.xhtml
+      class AuthType
+        Invalid = 0
+        None = 1
+        VNC = 2
+        # RealVNC has 3-15 registered
+        RealVNC_3 = 3
+        RealVNC_4 = 4
+        RA2 = 5
+        RA2ne = 6
+        RealVNC_7 = 7
+        RealVNC_8 = 8
+        RealVNC_9 = 9
+        RealVNC_10 = 10
+        RealVNC_11 = 11
+        RealVNC_12 = 12
+        RealVNC_13 = 13
+        RealVNC_14 = 14
+        RealVNC_15 = 15
+        Tight = 16
+        Ultra = 17
+        TLS = 18
+        VeNCrypt = 19 # TigerVNC this is Plain, TLSNone, TLSVnc, TLSPlain, X509Vnc, X509Plain
+        GtkVncSasl = 20
+        MD5Hash = 21
+        ColinDeanXVP = 22
+        SecureTunnel = 23
+        IntegratedSSH = 24
+        ARD = 30
+        # Apple has 31-35 registered, but no details on what these are. 36 is not registered but seems to be used by Apple
+        Apple_31 = 31
+        Apple_32 = 32
+        Apple_33 = 33
+        Apple_34 = 34
+        MacOSX_35 = 35
+        AppleUnknown_36 = 36
+        # 116 has been observed in the wild in the following combo: Ultra, Unknown: 116, VNC
```

Comment:
I'm not seeing 116 below so...is this a typo or what is going on here as I don't see a correlation.

Reply:
These are a little difficult.  Documentation is like non-existent and the only way to test is to find one in the wild and take a guess at which setting it is.  The default case is to print all unknowns as "Unknown: #" which i think is better than starting a  "Unknown_#" list and having it print uglier and uglier. I just wanted to note there that it had been seen before and with what other things so maybe someone else who comes across it is able to cross reference to what it truly is.

Reply:
Sounds good to me, just wanted to make sure this wasn't a mistake and was intentional 👍 Change accepted.